ksql-cli/src/main/java/io/confluent/ksql/Ksql.java

```diff
@@ -16,42 +16,64 @@
-import io.confluent.ksql.cli.commands.Local;
-import io.confluent.ksql.cli.commands.Remote;
-import io.confluent.ksql.cli.commands.Standalone;
-import io.confluent.ksql.util.CommonUtils;
+import org.apache.kafka.streams.StreamsConfig;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Optional;
+import java.util.Properties;
+
+import io.confluent.ksql.cli.Cli;
+import io.confluent.ksql.cli.Options;
+import io.confluent.ksql.cli.console.JLineTerminal;
+import io.confluent.ksql.rest.client.KsqlRestClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import io.confluent.ksql.version.metrics.KsqlVersionCheckerAgent;
+import io.confluent.ksql.version.metrics.collector.KsqlModuleType;
 
 public class Ksql {
 
-  public static void main(String[] args) {
-    Runnable runnable = null;
-    com.github.rvesse.airline.Cli<Runnable> cli =
-        com.github.rvesse.airline.Cli.<Runnable>builder("Cli")
-            .withDescription("Kafka Query Language")
-            .withDefaultCommand(Help.class)
-            .withCommand(Local.class)
-            .withCommand(Remote.class)
-            .withCommand(Standalone.class)
-            .build();
-
-    try {
-      runnable = cli.parse(args);
-      runnable.run();
-    } catch (ParseException exception) {
-      if (exception.getMessage() != null) {
-        System.err.println(exception.getMessage());
-      } else {
-        System.err.println("Options parsing failed for an unknown reason");
-      }
-      System.err.println("See the help command for usage information");
-    } catch (Exception e) {
-      System.err.println(CommonUtils.getErrorMessageWithCause(e));
+  public static void main(String[] args) throws IOException {
+    final Options options = Options.parse(args);
+    if (options == null) {
+      System.exit(-1);
     }
-    if ((runnable != null) && !(runnable instanceof Standalone)) {
-      System.exit(0);
+
+    final Properties properties = loadProperties(options.getPropertiesFile());
+    final KsqlRestClient restClient =
+        new KsqlRestClient(options.getServer(), properties);
+
+    options.getUserNameAndPassword().ifPresent(
+        creds -> restClient.setupAuthenticationCredentials(creds.left, creds.right));
+
+    final KsqlVersionCheckerAgent versionChecker = new KsqlVersionCheckerAgent();
+    versionChecker.start(KsqlModuleType.REMOTE_CLI, properties);
+    try(final Cli cli = new Cli(options.getStreamedQueryRowLimit(),
+        options.getStreamedQueryTimeoutMs(),
+        restClient,
+        new JLineTerminal(options.getOutputFormat(), restClient))) {
+        cli.runInteractively();
     }
+  }
 
+  @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+  private static Properties loadProperties(final Optional<String> propertiesFile) {
+    final Properties properties = new Properties();
+    properties.put(StreamsConfig.APPLICATION_ID_CONFIG, KsqlConfig.KSQL_SERVICE_ID_DEFAULT);
```

Comment:
`StreamConfig. APPLICATION_ID_CONFIG` will be set by KSQL so no need to have it anymore.

Reply:
where?

Reply:
It's set in `PhysicalPlanBuilder.java`.